examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -0,0 +1,457 @@
+#
+# It is worth noting that 92 % of policyholders have zero claims, and if we
+# were to convert this problem into a binary classification task, it would be
+# significantly imbalanced.
+#
+# To evaluate the pertinence of the used metrics, we will consider as a
+# baseline a "dummy" estimator that constantly predicts the mean frequency of
+# the training sample.
+
+df_train, df_test = train_test_split(df, random_state=0)
+
+dummy = make_pipeline(
+    linear_model_preprocessor,
+    DummyRegressor(strategy='mean')
+)
+dummy.fit(df_train, df_train["Frequency"],
+          dummyregressor__sample_weight=df_train["Exposure"])
+
+
+def score_estimator(estimator, df_test):
+    """Score an estimator on the test set."""
+
+    y_pred = estimator.predict(df_test)
+
+    print("MSE: %.3f" %
+          mean_squared_error(df_test["Frequency"], y_pred,
+                             df_test["Exposure"]))
+    print("MAE: %.3f" %
+          mean_absolute_error(df_test["Frequency"], y_pred,
+                              df_test["Exposure"]))
+
+    # ignore non-positive predictions, as they are invalid for
+    # the Poisson deviance
+    mask = y_pred > 0
+    if (~mask).any():
+        warnings.warn("Estimator yields non-positive predictions for {} "
+                      "samples out of {}. These will be ignored while "
+                      "computing the Poisson deviance"
+                      .format((~mask).sum(), mask.shape[0]))
+
+    print("mean Poisson deviance: %.3f" %
+          mean_poisson_deviance(df_test["Frequency"][mask],
+                                y_pred[mask],
+                                df_test["Exposure"][mask]))
+
+
+print("Constant mean frequency evaluation:")
+score_estimator(dummy, df_test)
+
+##############################################################################
+#
+# We start by modeling the target variable with the least squares linear
+# regression model,
+
+ridge = make_pipeline(linear_model_preprocessor, Ridge(alpha=1.0))
+ridge.fit(df_train, df_train["Frequency"],
+          ridge__sample_weight=df_train["Exposure"])
+
+##############################################################################
+#
+# The Poisson deviance cannot be computed on non-positive values predicted by
+# the model. For models that do return a few non-positive predictions
+# (e.g. :class:`linear_model.Ridge`) we ignore the corresponding samples,
+# meaning that the obtained Poisson deviance is approximate. An alternative
+# approach could be to use :class:`compose.TransformedTargetRegressor`
+# meta-estimator to map ``y_pred`` to a strictly positive domain.
+
+print("Ridge evaluation:")
+score_estimator(ridge, df_test)
+
+##############################################################################
+#
+# Next we fit the Poisson regressor on the target variable,
+
+poisson = make_pipeline(
+    linear_model_preprocessor,
+    PoissonRegressor(alpha=1/df_train.shape[0], max_iter=1000)
```

Comment:
please add a comment about why the `/df_train.shape[0]`. Basically it's like setting \alpha=1 in Ridge as Ridge does not scales by n_samples the loss and therefore can use alpha=1. by default.